command/filemanager.go

```diff
@@ -53,3 +60,133 @@ func (fm *FileManager) CopyInlineFile(name string, fileContent pulumi.StringInpu
+// For example, if this function is called from ~/foo/test.go with remoteFolder="bar"
+// then the full path of the folder will be ~/foo/bar“.
+// This function returns the resources that can be used with `pulumi.DependsOn`.
+func (fm *FileManager) CopyRelativeFolder(relativeFolder string, remoteFolder string, opts ...pulumi.ResourceOption) ([]pulumi.Resource, error) {
+	fullPath, rootFolder, err := getFullPath(relativeFolder, 2)
+	if err != nil {
+		return nil, err
+	}
+
+	return fm.CopyFSFolder(fullPath, os.DirFS(rootFolder), relativeFolder, remoteFolder, opts...)
+}
+
+// CopyRelativeFile copies relative path to a remote path.
+// The relative path is defined in the same way as for `CopyRelativeFolder`.
+// This function returns the resource that can be used with `pulumi.DependsOn`.
+func (fm *FileManager) CopyRelativeFile(relativePath string, remotePath string, opts ...pulumi.ResourceOption) (pulumi.Resource, error) {
+	fullPath, _, err := getFullPath(relativePath, 2)
+	if err != nil {
+		return nil, err
+	}
+
+	return fm.CopyFile(fullPath, remotePath, opts...)
+}
+
+// CopyFSFolder copies recursively a local folder to a remote folder.
+// You may consider using `CopyRelativeFolder` which has a simpler API.
+func (fm *FileManager) CopyFSFolder(
+	resourceName string,
+	folderFS fs.FS,
+	folderPath string,
+	remoteFolder string,
+	opts ...pulumi.ResourceOption) ([]pulumi.Resource, error) {
+	useSudo := true
+	folderCommand, err := fm.CreateDirectory(resourceName, pulumi.String(remoteFolder), useSudo, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create a temporary folder: %v for resource %v", err, resourceName)
+	}
+
+	files, folders, err := readFilesAndFolders(folderFS, folderPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read files and folders for %v", folderPath)
+	}
+
+	var folderResources []pulumi.Resource
+	for _, folder := range folders {
+		destFolder, err := getDestinationPath(folder, folderPath)
+		if err != nil {
+			return nil, err
+		}
+		remotePath := path.Join(remoteFolder, destFolder)
+		resources, err := fm.CreateDirectory("createFolder-"+remotePath, pulumi.String(remotePath), useSudo, pulumi.DependsOn([]pulumi.Resource{folderCommand}))
+		if err != nil {
+			return nil, err
+		}
+		folderResources = append(folderResources, resources)
+	}
+
+	fileResources := []pulumi.Resource{}
+	for _, file := range files {
+		destFile, err := getDestinationPath(file, folderPath)
+
+		if err != nil {
+			return nil, err
+		}
+
+		fileContent, err := fs.ReadFile(folderFS, file)
+		if err != nil {
+			return nil, err
+		}
+		fileCommand, err := fm.CopyInlineFile(
+			resourceName+"-"+file,
+			pulumi.String(fileContent),
+			path.Join(remoteFolder, destFile),
+			useSudo,
+			pulumi.DependsOn(folderResources))
+		if err != nil {
+			return nil, err
+		}
+		fileResources = append(fileResources, fileCommand)
+	}
+
+	return fileResources, nil
+}
+
+// When copying foo/bar to /tmp to result folder is /tmp/bar
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
```suggestion
// When copying foo/bar to /tmp the result folder is /tmp/bar
```